### PR TITLE
#10565: Cleanup and Remove std function for bw, complex ops

### LINKED ISSUE #10565: Remove std::function for binary, Unary BW ops
Tracking Issue : #10564
List of ops to cover
PR Link : #10572 
- [x] Binary, Unary div_bw
- [x] Unary remainder_bw
- [x] Unary fmod_bw
- [x] Unary, Binary sub_bw
- [x] Unary, Binary gt_bw
- [x] Unary, Binary Assign_bw
- [x] Unary multigammaln_bw
- [x] lgamma_bw
- [x] fill_bw
- [x] hardsigmoid_bw
- [x] cos_bw
- [x] acosh_bw

### LINKED ISSUE #10568: Remove std::function for bw ops
Tracking Issue : #10564
List of ops to cover: #10510

- [x] Unary, binary add_bw
- [x] Unary, Binary eq_bw
- [x] Unary, binary mul_bw

### PR DESCRIPTION
**Pull request by Aswinmcw**
Work Done in this PR:

- Clean up OpHandlers for the following
     - Unary , Binary, Ternary, Complex Backward ops
     - Complex Forward ops
- Remove std::function for unary and binary backward ops ( Issue : #10565 , #10568 - Tracking Issue : #10564 )

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/10111293765

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward.hpp [hunk 1]
```diff
@@ -99,9 +94,40 @@ struct ExecuteBinaryBackwardFloat {
...
+//OpHandler_binary_bw_opt : get_function_binary_bw_opt
+template <BinaryBackwardOpType binary_backward_op_type>
+struct ExecuteBinaryBackwardOpt {
+
+    static inline std::vector<ttnn::Tensor> create_async_output_tensors(
+        const std::vector<Tensor> &input_tensors, const std::vector<std::optional<const Tensor>>& optional_inputs) {
+        const auto& input_tensor = input_tensors.at(0);
+        return {Tensor(operation::get_workers_for_op_output({input_tensor})),
+                                            Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    }
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward.hpp:111]: I am a bit unfamiliar with this. Why is this needed?

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp [hunk 2]
```diff
@@ -237,6 +237,86 @@ void bind_binary_backward_float_default(py::module& module, const binary_backwar
...
+            },
+            py::arg("grad_tensor"),
+            py::arg("input_tensor_a"),
+            py::arg("input_tensor_b"),
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("are_required_outputs") = std::vector<bool>{true, true},
+            py::arg("input_a_grad") = std::nullopt,
+            py::arg("input_b_grad") = std::nullopt}
+    );
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp:317]: Why is this needed? The only difference I see is with queue_id, but it is a kw_only arg with a default value

**Review by ayerofieiev-tt — COMMENTED** [review]
PR title should include ticket number

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by ayerofieiev-tt — CHANGES_REQUESTED** [review]
Please remove `create_async_output_tensors`

**Review by eyonland — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Pull request merged**